kv/api_version.go

```diff
@@ -0,0 +1,57 @@
+package kv
+
+import "bytes"
+
+type Mode int
+
+const (
+	ModeRaw = iota
+	ModeTxn
+)
+
+var (
+	ApiV2RawKeyPrefix = []byte("r")
+	ApiV2TxnKeyPrefix = []byte("x")
+	ApiV2RawEndKey    = []byte{ApiV2RawKeyPrefix[0] + 1}
+	ApiV2TxnEndKey    = []byte{ApiV2TxnKeyPrefix[0] + 1}
+)
+
+func GetV2Prefix(mode Mode) []byte {
+	switch mode {
+	case ModeRaw:
+		return ApiV2RawKeyPrefix
+	case ModeTxn:
+		return ApiV2TxnKeyPrefix
+	}
+	panic("unreachable")
+}
+
+func GetV2EndKey(mode Mode) []byte {
+	switch mode {
+	case ModeRaw:
+		return ApiV2RawEndKey
+	case ModeTxn:
+		return ApiV2TxnEndKey
+	}
+	panic("unreachable")
+}
+
+func BuildV2RequestKey(mode Mode, key []byte) []byte {
+	return append(GetV2Prefix(mode), key...)
+}
+
+func DecodeV2StartKey(mode Mode, key []byte) []byte {
```

Comment:
Suggest to combine the following methods as `DecodeV2KeyRange`. Then it would be easier to be used as `startKey, endKey := DecodeV2KeyRange(startKey, endKey)`.